src/Web/E-Games.Web/E-Games.Web/Controllers/AuthController.cs

```diff
@@ -0,0 +1,139 @@
+        private readonly SignInManager<ApplicationUser> _signInManager;
+        private readonly IEmailSender _emailSender;
+
+        public AuthController(UserManager<ApplicationUser> userManager,
+            SignInManager<ApplicationUser> signInManager,
+            IEmailSender emailSender)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _emailSender = emailSender;
+        }
+
+        [HttpPost("signIn")]
+        [AllowAnonymous]
+        public async Task<IActionResult> SignInAsync([FromBody] SignModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var validateModel = ValidateModel(model);
+            if (validateModel != null)
+            {
+                return validateModel;
+            }
+
+            var user = await _userManager.FindByEmailAsync(model.Email!);
+
+            if (user != null)
+            {
+                var result = await _signInManager.PasswordSignInAsync(user, model.Password!, isPersistent: false, lockoutOnFailure: false);
+
+                if (result.Succeeded)
+                {
+                    return Ok(new { Message = "Sign in successful" });
+                }
+            }
+
+            return Unauthorized(new { Message = "Invalid email or password" });
+        }
+
+        [HttpPost("signUp")]
+        [AllowAnonymous]
+        public async Task<IActionResult> SignUpAsync([FromBody] SignModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var validateModel = ValidateModel(model);
+            if (validateModel != null)
+            {
+                return validateModel;
+            }
+
+            var user = new ApplicationUser { UserName = model.Email, Email = model.Email };
+            var result = await _userManager.CreateAsync(user, model.Password!);
+
+            if (result.Succeeded)
+            {
+                var token = await _userManager.GenerateEmailConfirmationTokenAsync(user);
+
+                var callbackUrl = Url.Action(nameof(EmailConfirmAsync),
+                    "Auth", new { userId = user.Id, token = token },
+                    protocol: HttpContext.Request.Scheme);
+
+                await _emailSender.SendEmailAsync(user.Email!,
+                    "Confirm your email",
+                    $"Please confirm your account by <a href='{callbackUrl}'>clicking here</a>.");
+
+                return StatusCode(StatusCodes.Status201Created);
+            }
+
+            return BadRequest(new { Errors = result.Errors.Select(e => e.Description) });
+        }
+
+        [HttpGet("emailConfirm")]
+        [AllowAnonymous]
+        public async Task<IActionResult> EmailConfirmAsync(string userId, string token)
+        {
+            if (string.IsNullOrWhiteSpace(userId))
```

Comment:
You don't need to do this manually. ASP validates your request model based on types, so, unless your parameter is nullable, you will get BadRequest.
![image](https://github.com/george-pench/E-Games/assets/39170652/88d10296-2997-4f47-a346-67ba524957a8)

Please go through controllers and remove unnecessary validation